### PR TITLE
structured codegen - refactor - position mapping

### PR DESCRIPTION
**Pull request by thebluegreeny**
Refactoring of the position mapping during structured code generation in order to avoid manual calculation of positions

### CODE REVIEW

**Comment by ltfish** [comment]
The changes look good to me, but it seems that your changes broke two test cases. Can you please fix that?

**Comment by thebluegreeny** [comment]
Should be fine now. At least the checks passed.

File: angr/analyses/decompiler/structured_codegen.py [hunk 1]
```diff
@@ -109,6 +93,18 @@ def __init__(self):
         pass
 
     def c_repr(self, indent=0, posmap=None):
+
+        def mapper(chunks, posmap):
+            pos = 0
+            for s, obj in chunks:
+                if obj:
```
Comment (ltfish) [inline angr/analyses/decompiler/structured_codegen.py:100]: Are we going to miss the case where `obj == 0`? I prefer a more explicit check here, aka `if obj is not None:`.

**Comment by ltfish** [comment]
Everything else looks good to me. I'll hit merge as soon as you change that line. Thanks a lot for your effort!

**Comment by ltfish** [comment]
Again, thanks a lot for refactoring it :) I really appreciate your time and effort.

**Pull request merged**